### PR TITLE
test(cosh-ng): cover SIGKILL atomic writes

### LINKED ISSUE #2510: [cosh-ng] test_write_file_kill9_no_truncated: LLM timing race + assertion edge case
The test test_write_file_kill9_no_truncated fails intermittently. The 3s sleep before kill -9 is insufficient for LLM API call (30-90s) to reach write_file phase. Fix: increase sleep or use smarter wait, add robust assertion handling.

Nightly: nightly-20260814-020042, ECS: 47.239.57.210, Component: cosh-ng

### PR DESCRIPTION
**Pull request by kongche-jbw**
## Why

The nightly test for #2510 used a fixed three-second delay before SIGKILL, so provider latency could prevent the write path from being reached. The test needs an explicit synchronization point at the atomic replacement boundary.

## What changed

Adds a Linux-only subprocess regression test that waits for a pre-commit readiness marker before sending SIGKILL. It verifies that the existing target remains complete when the process dies after the temporary file is synced but before the atomic rename.

## Related issue

Closes #2510

## User / Agent impact

None. This is a deterministic test-only change with no production behavior change.

## Risk and compatibility

Low risk. The synchronization environment variables exist only inside the test process, and the test is Linux-only because it validates SIGKILL semantics.

## Validation

- `cargo fmt --all -- --check`
- `cargo test --locked -p cosh-core tool::atomic_file::tests -- --nocapture` — 9 passed
- `cargo clippy --locked -p cosh-core --tests -- -D warnings`
- `git diff --check`

## Documentation and rollback

No documentation changes are required. Revert commit `7f914ae5` to remove the regression test.

### CODE REVIEW

File: src/cosh-ng/crates/cosh-core/src/tool/atomic_file.rs [hunk 1]
```diff
@@ -457,6 +466,72 @@ mod tests {
...
+                let _ = child.wait();
+                panic!("atomic replacement child did not reach the pre-commit barrier");
+            }
+            thread::sleep(Duration::from_millis(10));
+        }
+
+        child.kill().expect("send SIGKILL to replacement child");
+        let status = child.wait().expect("wait for replacement child");
+
+        assert_eq!(status.signal(), Some(9), "child status: {status}");
```
[the remark this answers is not in the document]
Comment (kongche-jbw) [reply src/cosh-ng/crates/cosh-core/src/tool/atomic_file.rs:531]: 感谢建议。该测试仅在 Linux 上编译运行；std::process::Child::kill() 在 Unix 上发送 SIGKILL，而 Linux 的 SIGKILL 信号编号固定为 9。#2510 验证的正是 Kill-9 场景，因此允许其他信号或 None 反而会削弱回归断言。当前断言也会输出完整的 child status，故这里保留精确检查。

File: src/cosh-ng/crates/cosh-core/src/tool/atomic_file.rs [hunk 2]
```diff
@@ -457,6 +466,72 @@ mod tests {
...
+            let _ = replace_with_hooks(
+                &write_target,
+                b"complete replacement",
+                None,
+                |_, _| Ok(()),
+                |_, _| {
+                    fs::write(&ready, b"ready")?;
+                    loop {
+                        thread::park();
+                    }
```
[the remark this answers is not in the document]
Comment (kongche-jbw) [reply src/cosh-ng/crates/cosh-core/src/tool/atomic_file.rs:493]: 感谢指出。当前控制流已经区分这两类情况：栅栏建立前发生 panic 时，父进程会通过 try_wait() 检测到子进程提前退出并立即报错；栅栏文件成功写入后，回调中没有其他可失败操作，只会 park 等待父进程发送 SIGKILL。因此该路径不会把提前 panic 误判为正常 Kill-9，本轮保持实现不变。

File: src/cosh-ng/crates/cosh-core/src/tool/atomic_file.rs [hunk 3]
```diff
@@ -457,6 +466,72 @@ mod tests {
...
+        const CHILD_READY_ENV: &str = "COSH_ATOMIC_SIGKILL_CHILD_READY";
+        const TEST_NAME: &str =
+            "tool::atomic_file::tests::sigkill_before_commit_keeps_existing_target_complete";
+
+        if let Some(target) = std::env::var_os(CHILD_TARGET_ENV) {
+            let target = PathBuf::from(target);
+            let ready = PathBuf::from(
+                std::env::var_os(CHILD_READY_ENV).expect("child ready path is configured"),
+            );
+            let write_target = prepare_test_target(&target);
```
[the remark this answers is not in the document]
Comment (kongche-jbw) [reply src/cosh-ng/crates/cosh-core/src/tool/atomic_file.rs:482]: 感谢建议。若栅栏文件写入失败，错误会经 ? 返回，子进程随后退出；父进程在轮询中会通过 try_wait() 报告“exited before the barrier”，不会将其视为已到达栅栏，也不会产生测试假通过。这里确实可以进一步细分诊断文本，但不影响测试正确性，而且栅栏位于同一临时目录并由同一用户写入，因此本轮不增加额外协议。
